---
title: "Find a gene project"
author: "Xiaoyan Wang(A16454055)"
format: pdf
editor: visual
---

**\[Q1\] Tell me the name of a protein you are interested in. Include the species and the accession number. This can be a human protein or a protein from any other species as long as it's function is known. If you do not have a favorite protein, select human RBP4 or KIF11. Do not use beta globin as this is in the worked example report that I provide you with online.** Name: MHCclass I-related gene protein Accession: NP_001522 Species: Homo Sapiens

**\[Q2\] Perform a BLAST search against a DNA database, such as a database consisting of genomic DNA or ESTs. The BLAST server can be at NCBI or elsewhere. Include details of the BLAST method used, database searched and any limits applied (e.g. Organism).** Method: TBLASTN search against Notamacropus ESTs Database: est Species: Notamacropus (taxid:1960649)--\>(Notamacropus eugenii (taxid:9315))

![](images/clipboard-3949112645.png)

![](images/clipboard-2420170139.png)

Chosen match: Accession FY568721.1, a 915bp long mRNA from Notamacropus tammar hypothalamus cDNA clone

![](images/clipboard-472064439.png)

![](images/clipboard-3874763787.png)

**\[Q3\] Gather information about this “novel” protein. At a minimum, show me the protein sequence of the “novel” protein as displayed in your BLAST results from \[Q2\] as FASTA format (you can copy and paste the aligned sequence subject lines from your BLAST result page if necessary) or translate your novel DNA sequence using a tool called EMBOSS Transeq at the EBI. Don’t forget to translate all six reading frames; the ORF (open reading frame) is likely to be the longest sequence without a stop codon. It may not start with a methionine if you don’t have the complete coding region. Make sure the sequence you provide includes a header/subject line and is in traditional FASTA format. Here, tell me the name of the novel protein, and the species from which it derives. It is very unlikely (but still definitely possible) that you will find a novel gene from an organism such as S. cerevisiae, human or mouse, because those genomes have already been thoroughly annotated. It is more likely that you willdiscover a new gene in a genome that is currently being sequenced, such as bacteria or plants or protozoa.**

![](images/clipboard-2658275355.png)

**\[Q4\] Prove that this gene, and its corresponding protein, are novel. For the purposes of this project, “novel” is defined as follows. Take the protein sequence (your answer to \[Q3\]), and use it as a query in a blastp search of the nr database at NCBI. • If there is a match with 100% amino acid identity to a protein in the database, from the same species, then your protein is NOT novel (even if the match is to a protein with a name such as “unknown”). Someone has already found and annotated this sequence, and assigned it an accession number. • If the top match reported has less than 100% identity, then it is likely that your protein is novel, and you have succeeded. • If there is a match with 100% identity, but to a different species than the one you started with, then you have likely succeeded in finding a novel gene. • If there are no database matches to the original query from \[Q1\], this indicates that you have partially succeeded: yes, you may have found a new gene, but no, it is not actually homologous to the original query. You should probably start over.**

![](images/clipboard-3015642477.png)

![](images/clipboard-2390978979.png)

**\[Q5\] Generate a multiple sequence alignment with your novel protein, your original query protein, and a group of other members of this family from different species. A typical number of proteins to use in a multiple sequence alignment for this assignment purpose is a minimum of 5 and a maximum of 20 - although the exact number is up to you. Include the multiple sequence alignment in your report. Use Courier font with a size appropriate to t page width. Side-note: Indicate your sequence in the alignment by choosing an appropriate name for each sequence in the input unaligned sequence le (i.e. edit the sequence le so that the species, or short common, names (rather than accession numbers) display in the output alignment and in the subsequent answers below). The goal in this step is to create an interesting an alignment for building a phylogenetic tree that illustrates species divergence.**

| 

![](images/clipboard-497855757.png)

| a group of other members of this family from different species

![](images/clipboard-897082339.png)

![](images/clipboard-3616637139.png)

![](images/clipboard-1075220934.png) **\[Q6\] Create a phylogenetic tree, using either a parsimony or distance-based approach. Bootstrapping and tree rooting are optional. Use “simple phylogeny” online from the EBI or any respected phylogeny program (such as MEGA, PAUP, or Phylip). Paste an image of your Cladogram or tree output in your report.**

![](images/clipboard-3902447129.png)

**\[Q7\] Generate a sequence identity based heatmap of your aligned sequences using R. If necessary convert your sequence alignment to the ubiquitous FASTA format (Seaview can read in clustal format and “Save as” FASTA format for example). Read this FASTA format alignment into R with the help of functions in the Bio3D package. Calculate a sequence identity matrix (again using a function within the Bio3D package). Then generate a heatmap plot and add to your report. Do make sure your labels are visible and not cut at the gure margins.**

```{r}
library(bio3d)
#install.packages("pheatmap")
library(pheatmap)
```

```{r}
aligned <- read.fasta("muscle-I20241203-193226-0346-30336184-p1m.fa")
ident <- seqidentity(aligned)
```

```{r, fig.height=6,fig.width=7}
pheatmap(ident)
```

**\[Q8\] Using R/Bio3D (or an online blast server if you prefer), search the main protein structure database for the most similar atomic resolution structures to your aligned sequences. List the top 3 unique hits (i.e. not hits representing different chains from the same structure) along with their Evalue and sequence identity to your query. Please also add annotation details of these structures. For example include the annotation terms PDB identifier (structureId), Method used to solve the structure (experimentalTechnique), resolution (resolution), and source organism (source). HINT: You can use a single sequence from your alignment or generate a consensus sequence from your alignment using the Bio3D function consensus(). The Bio3D functions blast.pdb(), plot.blast() and pdb.annotate() are likely to be of most relevance for completing this task. Note that the results of blast.pdb() contain the hits PDB identifier (or pdb.id) as well as Evalue and identity. The results of pdb.annotate() contain the other annotation terms noted above. Note that if your consensus sequence has lots of gap positions then it will be better to use an original sequence from the alignment for your search of the PDB. In this case you could chose the sequence with the highest identity to all others in your alignment by calculating the row-wise maximum from your sequence identity matrix.**

```{r}
# too many "-" appeared. The Notamacropus rufogriseus sequence, which is from the same family as the new discovered gene in Notamacropus eugenii, was selected as an example
exp <- read.fasta("example.fasta")
top <- blast.pdb(exp)
top3 <- top$hit.tbl[1:3,]
ann <- pdb.annotate(top3$pdb.id)
combined <- merge(top3, ann, by.x = "pdb.id", by.y = "row.names")
correct_names <- c("pdb.id", "experimentalTechnique", "resolution", "source", "evalue", "identity")
matching_cols <- intersect(correct_names, colnames(combined))  # Ensure only existing columns are selected
combined <- combined[, match(matching_cols, colnames(combined))]
combined
```

**\[Q9\] Generate a molecular figure of one of your identified PDB structures using VMD. You can optionally highlight conserved residues that are likely to be functional. Please use a white or transparent background for your figure (i.e. not the default black). Based on sequence similarity. How likely is this structure to be similar to your “novel” protein?**

The three targets identified from the previous step is: 7EDO, 7RTD, and 8RBU. As the 7EDO from another species have a very low E-value, I decided to look in this PDB structure.

![](images/clipboard-1245687751.png)

The identity is 71% similar to the "novel" protein from Notamacropus eugenii. This is likely a type of classical MHC- class I molecule. Since MHC class I is a highly polymorphic gene, I would think the 71% similarity between Notamacropus eugenii and the 7EDO protein from Trichosurus vulpecula supports that the "novel" protein is also a type of MHC Class I molecule. Henceforce, considering the nature of MHC Class I, I think these two proteins are similar.

**\[Q10\] Perform a “Target” search of ChEMBEL ( https://www.ebi.ac.uk/chembl/ ) with your novel sequence. Are there any Target Associated Assays and ligand efficiency reported that may be useful starting points for exploring potential inhibition of your novel protein?**

For this sequence, there is a HLA antigen target that is similar to the novel sequence we are looking. HLA is the MHC Class I molecule in human.

<https://www.ebi.ac.uk/chembl/explore/target/CHEMBL2632#NameAndClassification>

There are 3 related binding assays listed here.

<https://www.ebi.ac.uk/chembl/explore/assays/STATE_ID:yG22iojPssyMVh9TAuiLXQ%3D%3D>

Two of them are studying the binding affinity through inhibiting a ratiolabled standard peptide(FLPSDYFPSV). The novel protein, which is also a type of MHC Class I or MHC Class I with antigen presented, could be also inhibited through similar way targeting the antigen-binding site.
